### DISCUSSION TITLE
Flatpak release

### DISCUSSION THREAD

**Discussion by darkdragon-001** [input-leap/input-leap#1652] [Development of Input Leap]
### What happened?

With libei integration for Wayland in #1594 in combination with [portals](https://gitlab.freedesktop.org/libinput/libei/-/issues/1), it should be possible to run input-leap as Flatpak.
@whot do you agree?

### Version

From Git HEAD or commit (specify below)

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### What OS versions are you using?

Ubuntu

**Comment by shymega** [comment]
In future, this should be kept to a DIscussion. A Flatpak package is under way on my personal account. However, it can't be PR'd until v3.0.0 is released.

**Comment by whot** [comment]
> it should be possible to run input-leap as Flatpak.

yep, that's the whole point of libei (and the complicated architecture) - being able to run inputleap as flatpak on both server and client side. *however*: until the xdg-desktop-portal changes are merged and compositors have the support for it there's no point because it would be talking into the void.

**Comment by orowith2os** [comment]
If help is needed when the portals are merged, I'd love to poke around. @shymega you mentioned having a Flatpak package somewhere, can you link me to it? Thanks.

**Reply by shymega** [reply]
Here it is: https://github.com/shymega/flathub/tree/add/flatpak/input-leap

**Reply by orowith2os** [reply]
Maybe upstream the metainfo.xml file here, as that's not Flatpak-specific, and maintaining it here can give users more info on what happens in an update. Otherwise, unless you manually maintain it, you end up with lacking version upgrade details.

**Reply by shymega** [reply]
Yeah, I derived it from the Barrier Flatpak.

I could move it. I just don't know if the Flathub team would prefer that or not. I would rather keep all Flatpak-specific files in the Input Leap repo, just like Snap, so we can build from one repository.

**Reply by whot** [reply]
> I just don't know if the Flathub team would prefer that or not

ideally you'd just have the flatpak-specific `.json` or `.yaml` build file in the flatub repo (and patches, if needed) and everything else is upstream. The appdata file is independent of flatpak and should thus be upstream.

**Reply by shymega** [reply]
Replying here. (@orowith2os you may want to reply inside this 'bubble', so it's all in the same thread. I'll quote your previous reply in this thread.)


> ideally you'd just have the flatpak-specific .json or .yaml build file in the flatub repo (and patches, if needed) and everything else is upstream. The appdata file is independent of flatpak and should thus be upstream.

@whot 

Yeah, I can see that being a better option.

> The Flatpak manifest could, maybe, at some point be moved upstream too, with workflows to generate bundles for it.
> If the Flatpak would ever be official, this would be useful. That should probably wait until everything else clicks into place, though.

@orowith2os 

I would be interested to know how to move the Flatpak manifest in this repo. I've been looking into how to do that with another project of mine, but I had no response on the Flatpak Matrix channel. 

On the other hand, maybe we should just take advantage of the Flathub infrastructure?

**Reply by whot** [reply]
> On the other hand, maybe we should just take advantage of the Flathub infrastructure? 

I strongly encourage that, it'll make the flatpak easier to find, etc. Speaking from my experience maintaining the [piper flathub repo](https://github.com/flathub/org.freedesktop.Piper) it's literally just a json file in a github repo that rebuilds the flathub application on git push. Can't get much easier than that and provides the best discoverability.

**Reply by shymega** [reply]
> > On the other hand, maybe we should just take advantage of the Flathub infrastructure?
> 
> I strongly encourage that, it'll make the flatpak easier to find, etc. Speaking from my experience maintaining the [piper flathub repo](https://github.com/flathub/org.freedesktop.Piper) it's literally just a json file in a github repo that rebuilds the flathub application on git push. Can't get much easier than that and provides the best discoverability.

Fair enough. And also, it means we don't have to worry ourselves with CI or infra.

I can move the manifest over tomorrow to this repo.

**Comment by orowith2os** [comment]
The Flatpak manifest could, maybe, at some point be moved upstream too, with workflows to generate bundles for it. 

If the Flatpak would ever be official, this would be useful. That should probably wait until everything else clicks into place, though.

**Reply by whot** [reply]
> If the Flatpak would ever be official, this would be useful. That should probably wait until everything else clicks into place, though.

ftr, you can already ship inputleap as flatpak, you just have to open up the x11 hole in the sandbox (and others, i guess). Which removes most of the sandboxing features of course but that's the point of those sandbox holes - you can start with the sandbox being a sieve and plug the holes one by one as the code-base adjusts.

So in your case the benefit of a flatpak isn't that the app now runs super-secure but rather that you can ask users "can you reproduce it in this flatpak" and you have a single reference code base.

**Reply by shymega** [reply]
Yeah. I can't ship it yet, want to wait until v3. We have a few more bugs to iron out.

**Reply by darkdragon-001** [reply]
Flathub also offers a beta channel 😉

**Reply by shymega** [reply]
That's true. I suppose we could release it early on Flatpak. Let me discuss with Povilas.

**Comment by orowith2os** [comment]
Is there anything special now that's required for Flatpak? The libei stuff has landed, and once GNOME 45 stable lands it should be safe to release it. 

Is there a checklist for the next Input Leap release, or would it be fine to ship -git with Flathub Beta, or a standalone repo/bundles somewhere?

**Reply by shymega** [reply]
@orowith2os There's no checklist, but I have made a milestone. I've heard that Flathub don't approve of shipping -git HEAD with the Beta channel.

I would personally make a Beta release from HEAD for now. That way people can get started, quickly. I've moved my Flathub repo to the organisation, so if you'd like to make any PRs for the AppData XML, and the build manifest, please do! The build manifest in `flathub` on the IL org, is mine, but there's also one inside the main repo. I think the latter is a better fit.

That way we can get a Flatpak going. Albeit Git only for now.

**Reply by karypid** [reply]
Is there a flatpak that I can use to test it? The AUR no longer compiles and it seems there is no flatpak released (even in beta) currently. Would like to give this a spin if at all possible.

**Comment by Obsessee** [comment]


**Reply by p12tic** [reply]
@shymega Could you tell me what would be the manual release process? We can automate the process later, for now manual process is good enough.
